### PR TITLE
Support registering system Ruby

### PR DESCRIPTION
**Pull request by p0deje**
Allows to avoid downloading/installing Ruby interpreter at the expense of losing hermeticity and no Ruby version check.

```bazel
# WORKSPACE
load("@rules_ruby//ruby:deps.bzl", "rb_register_toolchains")

rb_register_toolchains(
    version = "system",
)
```

@kigster Does it look reasonable for your case of migrating from Bazelruby?

### CODE REVIEW

File: ruby/private/download.bzl [hunk 1]
```diff
@@ -167,7 +96,13 @@ def _install_via_ruby_build(repository_ctx):
     if result.return_code != 0:
         fail("%s\n%s" % (result.stdout, result.stderr))
 
-_rb_download = repository_rule(
+def _symlink_system_ruby(repository_ctx):
+    ruby = repository_ctx.which("ruby")
```
Comment (alexeagle) [inline ruby/private/download.bzl:100]: I think this matches what @thesayyn has done in some other toolchains 👍🏻

File: ruby/private/toolchain.bzl [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+def rb_register_toolchains(name = DEFAULT_RUBY_REPOSITORY, version = None, register = True, **kwargs):
+    """
+    Register a Ruby toolchain and lazily download the Ruby Interpreter.
+
+    * _(For MRI on Linux and macOS)_ Installed using [ruby-build](https://github.com/rbenv/ruby-build).
+    * _(For MRI on Windows)_ Installed using [RubyInstaller](https://rubyinstaller.org).
+    * _(For JRuby on any OS)_ Downloaded and installed directly from [official website](https://www.jruby.org).
+    * _(For TruffleRuby on Linux and macOS)_ Installed using [ruby-build](https://github.com/rbenv/ruby-build).
+    * _(For "system") System Ruby is used. Please note that builds are not hermetic in this case.
```
Comment (alexeagle) [inline ruby/private/toolchain.bzl:16]: I would say "ruby found on the PATH" to be a bit more precise about what a "System Ruby" is

**Review by alexeagle — APPROVED** [review]

**Comment by kigster** [comment]
I am not sure — if by "system" you mean the default Ruby installed by OS-X, then unfortunately it's not the same. 

The old rules_ruby supported the local `rbenv` instalation which typically resides under `$HOME/.rbenv/versions/<version>`.  There may be multiple versoins there, and each project is allowed to set required ruby version by including the `.ruby-version` file, which makes `rbenv` automatically switch to that version.

A mono-repo may have several sub-projects that reference different ruby versions. While this is not something I would recommend anyone to do in practive, the reality is that this does happen quite often within large companies.

Therefore I would suggest differentiating between `system` ruby, which on OS-X is the pre-installed ruby (and which, unfortunately, is installed on a read-only file system, and does not permit adding any additional gems, and is therefore largely unusable unless you are relying on stdlib and nothing else), and `rbenv` or even `rvm` rubies, which are in installed under `~/.rbenv` or `~/.rvm` folders respectively.

**Comment by p0deje** [comment]
As mentioned in the documentation, "system" means Ruby found on the PATH. If it's registered by rbenv, then it will be picked up respectfully.

**Pull request merged**